client/verta/tests/unit_tests/pipeline/test_registered_pipeline.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the RegisteredPipeline class
+"""
+
+import pytest
+from hypothesis import given, HealthCheck, settings
+
+from tests.unit_tests.strategies import mock_pipeline_definition
+from verta.pipeline import RegisteredPipeline
+
+
+def test_copy_graph(
+    make_mock_pipeline_graph, make_mock_registered_model_version
+) -> None:
+    """Test that the graph of a RegisteredPipeline can be copied"""
+    graph = make_mock_pipeline_graph()
+    pipeline = RegisteredPipeline(
+        pipeline_graph=graph,
+        registered_model_version=make_mock_registered_model_version(),
+    )
+    copied_graph = pipeline.copy_graph()
+    assert copied_graph.steps == graph.steps  # same steps
+    assert copied_graph != graph  # different objects
+
+
+@given(pipeline_definition=mock_pipeline_definition())
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    deadline=None,
+)
+def test_log_pipeline_definition_artifact(
+    pipeline_definition,
+    mocked_responses,
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+) -> None:
+    """
+    Verify the expected sequence of calls when a pipeline definition
+    is logged as an artifact to the pipeline's model version.
+    """
+    pipeline = RegisteredPipeline(
+        pipeline_graph=make_mock_pipeline_graph(),
+        registered_model_version=make_mock_registered_model_version(),
+    )
+    # Fetch the model
+    mocked_responses.get(
+        f"https://test_socket/api/v1/registry/model_versions/{pipeline.id}",
```

Comment:
We should get `"https"` and `"test_socket"` from `pipeline._registered_model_version._conn.scheme` and `socket` instead of hard-coding them into the test like this.

If we ever change the test fixture, we shouldn't have to go and debug every test where we've hard-coded its values.

Reply:
Good call.  Updated to pull the scheme and socket dynamically